### PR TITLE
Add a benchmark for for `unmultiply_inplace` in `pixels`

### LINKED ISSUE #33552: Move `unmultiply_inplace` to pixels crate and optimize it
https://github.com/servo/servo/blob/7fdaccde5501283708503e9a64efa578ac9ae7f7/components/canvas/webgl_thread.rs#L2841-L2848
while more performant (not tested just assumed because no fp operations) is in wrench:
https://github.com/servo/webrender/blob/bd7704a7091c51593cef9a3fe2ba86eb7e26359e/wrench/src/premultiply.rs#L21

we should move function to components/pixels, write benchmark and see if wrench version performs better.

### PR DESCRIPTION
**Pull request by sagudev**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes partial fix #33552 
- [x] These changes do not require tests because they are tests

### CODE REVIEW

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/pixels/benches.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+use criterion::*;
+
+fn create_data(pixels: usize) -> Vec<u8> {
+    let mut v = vec![];
+    for i in 0..=pixels {
+        let i = (i % 255) as u8;
+        v.extend_from_slice(&[i, i, i, i]);
+    }
+    v
+}
```
[the remark this answers is not in the document]
Comment (sagudev) [reply components/pixels/benches.rs:14]: I completely forgot `flatten` exist.

File: components/pixels/benches.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use criterion::*;
+
+fn create_data(pixels: usize) -> Vec<u8> {
```
[the remark this answers is not in the document]
Comment (sagudev) [reply components/pixels/benches.rs:7]: I am not really sure given that we just need number of pixels and throughput benchmarks that we could do in the future works with only one parameter.

[the remark this answers is not in the document]
Comment (mrobinson) [reply components/pixels/benches.rs:7]: Let's change the name of this parameter to `number_of_pixels` then and not use a multiplication below as that makes it seem like we are testing a width and a height.

**Review by mrobinson — APPROVED** [review]
Thanks!

**Pull request merged**